docs/docsite/rst/dev_guide/developing_collections_specifics.rst

```diff
@@ -0,0 +1,130 @@
+
+* If are not using this tool, include the properly formatted ``changelog.yaml`` file  into your collection. See the `changelog.yaml format <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md>`_ for details.
+
+* Add a link to own changelogs or release notes in any format by opening an issue at https://github.com/ansible-community/ansible-build-data/ with the HTML link to that information.
+
+.. note::
+
+  For the first two options, Ansible pulls the changelog details from Galaxy so your changelogs must be included in the collection version on Galaxy that is included in the upcoming Ansible release.
+
+Leveraging optional module utilities in collections
+===================================================
+
+Optional module utilities let you adopt the latest features from the most recent ansible-core release in your collection-based modules without breaking your collection on older Ansible versions. With optional module utilities, you can leverage the latest features when running against the latest versions, while still providing fallback behaviors when running against older versions.
+
+This implementation, widely used in Python programming, wraps optional imports in conditionals or defensive `try/except` blocks, and implements fallback behaviors for missing imports. Ansible's module payload builder supports these patterns by treating any module_utils import nested in a block (e.g., `if`, `try`) as optional. If the requested import cannot be found during the payload build, it is simply omitted from the target payload and assumed that the importing code will handle its absence at runtime. Missing top-level imports of module_utils packages (imports that are not wrapped in a block statement of any kind) will fail the module payload build, and will not execute on the target.
+
+For example, the `ansible.module_utils.common.respawn` package is only available in Ansible 2.11 and higher. The following module code would fail during the payload build on Ansible 2.10 or earlier (as the requested Python module does not exist, and is not wrapped in a block to signal to the payload builder that it can be omitted from the module payload):
+
+.. code-block:: python
+
+   from ansible.module_utils.common.respawn import respawn_module
+
+By wrapping the import statement in a ``try`` block, the payload builder will omit the Python module if it cannot be located, and assume that the Ansible module will handle it at runtime:
+
+.. code-block:: python
+
+   try:
+       from ansible.module_utils.common.respawn import respawn_module
+   except ImportError:
+       respawn_module = None
+   ...
+   if needs_respawn:
+       if respawn_module:
+           respawn_module(target)
+       else:
+           module.fail_json('respawn is not available in Ansible < 2.11, ensure that foopkg is installed')
+
+The optional module_utils behavior also applies to module imports from collections.
```

Comment:
Reading this back, it could be incorrectly interpreted that we're allowing imports *of* Ansible modules from collections. Maybe reword to something like:

```suggestion
The optional import behavior also applies to module_utils imported from collections.
```

Reply:
Thanks @nitzmahone for the review. I changed the line number in a different edit, so the suggestion is "outdated" and I can't commit it from here. I implemented this exact change in 4b03edcf8d8ccdc1d480e86e457c70141e051c9a.